### PR TITLE
ANDROAPP-5464-mobile-ui-Input

### PR DESCRIPTION
**Pull request by xavimolloy**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt [hunk 1]
```diff
@@ -16,36 +16,74 @@ import androidx.compose.runtime.setValue
...
 @Composable
-fun SimpleTextInputField(
-    enabled: Boolean = true
+fun BasicInput(
+    helper: String? = null,
+    enabled: Boolean = true,
+    helperStyle: InputStyle = InputStyle.NONE
 ) {
-    var text by remember { mutableStateOf("") }
+    var text by remember { mutableStateOf("Input") }
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt:62]: We should about hard coded text

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Text.kt [hunk 2]
```diff
@@ -47,5 +55,78 @@ internal fun InputShellLabelText(
...
+    Text(
+        text,
+        modifier = modifier,
+        color = textColor,
+        style = MaterialTheme.typography.titleSmall,
+        textAlign = TextAlign.Start
+    )
+}
+
+class PrefixTransformation(val prefix: String) : VisualTransformation {
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Text.kt:67]: shouldn't these class be in specific files?

Comment (andresmr) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Text.kt:67]: also if we don't want to expose them, we can create a package inside components called internal and place them there with the internal annotation?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt (hunk 1 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/InputField.kt:62]: yes you are right I set it up like that for the example and it passe me by

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Text.kt (hunk 2 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/Text.kt:67]: Good idea! I'll refactor it

**Review by Balcan — APPROVED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/Utils.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.TransformedText
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.sp
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+
+internal class PrefixTransformation(val prefix: String) : VisualTransformation {
```
Comment (DavidAparicioAlbaAsenjo) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/Utils.kt:12]: prefix should be private?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/Utils.kt
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/internal/Utils.kt:1]: Could this be more specific: StringUtils?

**Review by andresmr — APPROVED** [review]

**Review by DavidAparicioAlbaAsenjo — APPROVED** [review]

**Pull request merged**